packages/@tinacms/toolkit/src/plugins/branch-switcher/BranchSwitcherLegecy.tsx

```diff
@@ -0,0 +1,358 @@
+import * as React from 'react'
+import { BranchSwitcherProps, Branch } from './types'
+import { useBranchData } from './BranchData'
+import { BaseTextField, Input } from '../../packages/fields'
+import { Button } from '../../packages/styles'
+import { LoadingDots } from '../../packages/form-builder'
+import {
+  BiError,
+  BiGitBranch,
+  BiPlus,
+  BiRefresh,
+  BiSearch,
+} from 'react-icons/bi'
+import { GrCircleQuestion } from 'react-icons/gr'
+import { MdArrowForward, MdOutlineClear } from 'react-icons/md'
+import { AiFillWarning } from 'react-icons/ai'
+import { FaSpinner } from 'react-icons/fa'
+import { useCMS } from '../../packages/react-core'
+
+type ListState = 'loading' | 'ready' | 'error'
+
+export function formatBranchName(str: string): string {
+  const pattern = /[^/\w-]+/g // regular expression pattern to match invalid special characters
+  const formattedStr = str.replace(pattern, '') // remove special characters
+  return formattedStr.toLowerCase()
+}
+
+export const BranchSwitcherLegacy = ({
+  listBranches,
+  createBranch,
+  chooseBranch,
+}: BranchSwitcherProps) => {
+  const cms = useCMS()
+  const isLocalMode = cms.api?.tina?.isLocalMode
+  const [listState, setListState] = React.useState<ListState>('loading')
+  const [branchList, setBranchList] = React.useState([] as Branch[])
+  const { currentBranch } = useBranchData()
+  const initialBranch = React.useMemo(() => currentBranch, [])
+  // when modal closes, refresh page is currentBranch has changed
+  React.useEffect(() => {
+    return () => {
+      if (initialBranch != currentBranch) {
+        window.location.reload()
+      }
+    }
+  }, [currentBranch])
+
+  const handleCreateBranch = React.useCallback((value) => {
```

Comment:
<img width="820" alt="Screenshot 2023-06-21 at 10 21 04 AM" src="https://github.com/tinacms/tinacms/assets/5414297/28b72368-b0e7-47a9-8806-067512293362">

The prefix here doesn't actually get applied to the branch, testing it out with input "hello", just creates a branch named "hello" instead of "tina/hello"

> Note that this _does_ work as expected when prompted after saving

<img width="728" alt="Screenshot 2023-06-21 at 10 27 46 AM" src="https://github.com/tinacms/tinacms/assets/5414297/c7fa1220-9760-4ab5-a7f3-70f626ef96c9">

Reply:
@jeffsee55 Good catch, should be an easy fix.